pkg/ingester/ingester.go

```diff
@@ -2262,7 +2263,7 @@ func (i *Ingester) trackInflightQueryRequest() (func(), error) {
 
 	i.maxInflightQueryRequests.Track(i.inflightQueryRequests.Inc())
 
-	if i.resourceBasedLimiter != nil {
+	if i.resourceBasedLimiter != nil && requestmeta.RequestSourceFromContext(ctx) == requestmeta.SourceApi {
```

Comment:
nit: what do you think of adding helper functions in requestmeta itself? ie. `requestmeta.RequestFromApi(ctx)` and `requestmeta.RequestFromRuler(ctx)`